pkg/github/config.go

```diff
@@ -43,28 +46,55 @@ type Config struct {
 	GitHubRunAttempt        int64
 	GitHubJob               string
 	GitHubPullRequestNumber int
+	GitHubPullRequestBody   string
 	GitHubSHA               string
 }
 
+type configDefaults struct {
+	Owner             string
+	Repo              string
+	PullRequestNumber int
+	PullRequestBody   string
+}
+
 func (c *Config) RegisterFlags(set *cli.FlagSet) {
+	d := &configDefaults{}
+
 	githubContext, _ := githubactions.New().Context()
-	owner, repo := githubContext.Repo()
-	prNumber, _ := githubContext.Event["number"].(int)
+	d.Owner, d.Repo = githubContext.Repo()
+
+	data, _ := json.Marshal(githubContext.Event) //nolint:errchkjson //Shouldnt affect defaults
+
+	if githubContext.EventName == "pull_request" {
+		var event github.PullRequestEvent
+		if err := json.Unmarshal(data, &event); err == nil {
+			d.PullRequestNumber = event.GetNumber()
+			d.PullRequestBody = event.GetPullRequest().GetBody()
+		}
+	}
 
 	f := set.NewSection("GITHUB OPTIONS")
 
+	f.StringVar(&cli.StringVar{
+		Name:   "guardian-github-token",
+		EnvVar: "GUARDIAN_GITHUB_TOKEN",
+		Target: &c.GuardianGitHubToken,
+		Usage:  "The GitHub access token for Guardian to make GitHub API calls.",
```

Comment:
Can't effectively tell the difference between this one and the one below. I suppose they are used for different GH APIs?

Reply:
See this comment: https://github.com/abcxyz/guardian/pull/386#discussion_r1729112429

Reply:
gave better description